### PR TITLE
types: Handle IPv6 literals correctly in port bindings

### PR DESCRIPTION
**Pull request by euanh**
Use net.splitHostPort() instead of our own logic in func (p *PortBinding)
FromString(s string) error.   This means that IPv6 literals, including
IPv4 in IPv6 literals, can now be parsed from the string form of
PortBindings.   Zoned addresses do not work - net.splitHostPort() parses
them but net.ParseIP() cannot and returns an error.   This is ok because
we do not have a slot to store the zone name in PortBinding anyway.

Signed-off-by: Euan Harris <euan.harris@docker.com>

### CODE REVIEW

File: types/types_test.go [hunk 1]
```diff
@@ -26,21 +26,66 @@ func TestTransportPortConv(t *testing.T) {
...
-	if err := rc.FromString(sform); err != nil {
-		t.Fatal(err)
-	}
-	if !pb.Equal(rc) {
-		t.Fatalf("FromString() method failed")
+	for _, in := range input {
+		rc := new(PortBinding)
+		err := rc.FromString(in.sform)
+		if in.shouldFail {
+			if err == nil {
```
Comment (fcrisciani) [inline types/types_test.go]: you can use the require package `require.NoError(t, err)` and `require.Error(t, err)`
it's more compact

Comment (euanh) [reply types/types_test.go]: Thanks.   I've seen that used elsewhere but I was following the existing style in this file.

Do we have a coding style guide?   Should we be using testify for all new tests (and converting existing ones over as we update them)?

Comment (fcrisciani) [reply types/types_test.go]: I was suggesting mainly for brevity and readability. Don't think is worth going back and update all the tests, maybe if we spend time updating some then we can simplify it, but not mandatory

**Comment by euanh** [comment]
@fcrisciani Ping!

**Comment by selansen** [comment]
LGTM

**Pull request merged**